src/Serialization/Rings.jl

```diff
@@ -642,26 +678,23 @@ function load_object(s::DeserializerState, ::Type{MPolyQuoRing})
 end
 
 ### Serialization of Monomial orderings
-@register_serialization_type MonomialOrdering
+@register_serialization_type MonomialOrdering uses_params
 
 function save_object(s::SerializerState, o::MonomialOrdering)
   save_data_dict(s) do
-    save_typed_object(s, base_ring(o), :ring)
     save_typed_object(s, o.o, :internal_ordering) # TODO: Is there a getter for this?
     if isdefined(o, :is_total)
-      save_typed_object(s, o.is_total, :is_total)
+      save_object(s, o.is_total, :is_total)
     end
   end
 end
 
-function load_object(s::DeserializerState, ::Type{MonomialOrdering})
-  the_ring = load_typed_object(s, :ring)
-  the_ordering = load_typed_object(s, :internal_ordering)
-  result = MonomialOrdering(the_ring, the_ordering)
+function load_object(s::DeserializerState, ::Type{MonomialOrdering}, ring::MPolyRing)
+  ord = load_typed_object(s, :internal_ordering)
+  result = MonomialOrdering(ring, ord)
 
-  println("need to come back to this")
```

Comment:
This looks weird. But what I do remember is that we did not make a serious attempt at serializing all monomial orderings. I only did one, the default ordering for `MPolyQuoRing`s. That was all I needed at that point.

Reply:
true, it seems possible that we left this print in the wrong place. We currently can only do the 
Orderings.SymbOrdering,  I just added a comment on top of orderings serialization that is probablu more useful. Since the SymbOrderings are all I needed for some first examples of Ideal Gens and this loading of Monomial orderings is working for them. I think this is fine, essentially this might break when adding more orderings, but this should be easy to find where the fix needs to happen when faced with a backtrace

Reply:
Depends who you have in mind. What I would probably do is to leave an error message along the lines "serialization of monomial orderings of type $(typeof(this_ordering)) not implemented; consult docs/src/how_to_do_my_serialization.md for more information."

Reply:
This is a good point, and reminds that i need to fix up the error messages for when types aren't implemented. I'll do this in the another pull request.